src/server.c

```diff
@@ -1356,6 +1373,9 @@ int serverCron(struct aeEventLoop *eventLoop, long long id, void *clientData) {
             }
     }
 
+    /* Pause clients on write as long as OOM and pending lazy-free jobs in background */
+    ClientWritePauseDuringOOM();
```

Comment:
I presume the main functionality of this call is to unpause the clients if there is no traffic, but the comment indicates the opposite.

Reply:
The main functionality of this function is to throttle (pause-unpause) clients-write operation during OOM. Maybe It should be named `ThrottleClientWritePauseDuringOOM`. Maybe you have a better name?

Reply:
Doing pause-unpause per command will block main thread to serve other clients. 
Maybe have it called from beforeSleep and serverCron?

Reply:
Added to beforeSleep()

Reply:
come to think of it, we can remove the serverCron one, since after a cron timer wakeup, we go through beforeSleep anyway.